spark/v2.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -196,4 +198,40 @@ protected static Object convertConstant(Type type, Object value) {
     }
     return value;
   }
+
+  protected void findMoreCompoundConstants(Schema tableSchema, Map<Integer, ?> idToConstant) {
+    List<Types.NestedField> columns = tableSchema.columns();
+
+    for (Types.NestedField nestedField : columns) {
+      visitCompoundTypesByDFS(nestedField, (Map<Integer, Object>) idToConstant);
+    }
+  }
+
+  protected void visitCompoundTypesByDFS(
+          Types.NestedField nestedField,
+          Map<Integer, Object> idToConstant) {
```

Comment:
I'm not sure that this should alter the incoming map. It would be better to build a new constants map with the extra fields.